pkg/operator/status_test.go

```diff
@@ -102,3 +104,84 @@ func TestOperatorStatusProgressing(t *testing.T) {
 		assert.True(t, condition.LastTransitionTime.Equal(&conditionAfterAnotherSync.LastTransitionTime), "test-case %v expected LastTransitionTime not to be updated if condition state is same", i)
 	}
 }
+
+func TestGetOrCreateClusterOperator(t *testing.T) {
+	var namespace = "openshift-machine-api"
+	var conditions = []osconfigv1.ClusterOperatorStatusCondition{
+		{
+			Type:               "Available",
+			Status:             "true",
+			Reason:             "",
+			Message:            "",
+			LastTransitionTime: metav1.NewTime(time.Now()),
+		},
+	}
+	testCases := []struct {
+		existingCO *osconfigv1.ClusterOperator
+		expectedCO *osconfigv1.ClusterOperator
+	}{
+		{
+			existingCO: nil,
+			expectedCO: &osconfigv1.ClusterOperator{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: clusterOperatorName,
+				},
+				Status: osconfigv1.ClusterOperatorStatus{
+					RelatedObjects: []osconfigv1.ObjectReference{
+						{
+							Group:    "",
+							Resource: "namespaces",
+							Name:     namespace,
+						},
+					},
+				},
+			},
+		},
+		{
+			existingCO: &osconfigv1.ClusterOperator{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: clusterOperatorName,
+				},
+				Status: osconfigv1.ClusterOperatorStatus{
+					Conditions: conditions,
+				},
+			},
+			expectedCO: &osconfigv1.ClusterOperator{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: clusterOperatorName,
+				},
+				Status: osconfigv1.ClusterOperatorStatus{
+					RelatedObjects: []osconfigv1.ObjectReference{
+						{
+							Group:    "",
+							Resource: "namespaces",
+							Name:     namespace,
+						},
+					},
+					Conditions: conditions,
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		var osClient *fakeconfigclientset.Clientset
+		if tc.existingCO != nil {
+			osClient = fakeconfigclientset.NewSimpleClientset(tc.existingCO)
+		} else {
+			osClient = fakeconfigclientset.NewSimpleClientset()
+		}
+		optr := Operator{
+			osClient:  osClient,
+			namespace: namespace,
+		}
+
+		co, err := optr.getOrCreateClusterOperator()
+		if err != nil {
+			t.Errorf("unexpected error: %v", err)
```

Comment:
This should probably be a call to `Fatal`. Can the test continue of `co` is nil?